Only generate docs once per run when executing API tests

andrew-oc opened the pull request:
### Description

This is a simple change to make docfx run only once for any single test run. 

### Notes

This could have been done as a one time test fixture setup but we don't generate docs for all API tests and we don't want to pay the cost of docs generation for tests that don't care about it.

### Checklist

Before review:

- [ ] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - FogBugz ticket attached, example `([case %number%](https://issuetracker.unity3d.com/issues/...))`.
    - FogBugz is marked as "Resolved" with *next* release version correctly set.
- [ ] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [ ] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

During merge:

- [ ] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

andrew-oc committed:
Only generate docs once per run

API verfication tests would run docfx five times per test run! Now they are run once per test run.

jimon commented on Assets/Tests/InputSystem/APIVerificationTests.cs:
```diff
@@ -412,9 +416,17 @@ private static bool IgnoreMethodForDocs(MethodDefinition method)
-    ////TODO: move this to a fixture setup so that it runs *once* for all API checks in a test run
     private static string GenerateDocsDirectory(out string log)
     {
+        // only generate the docs once for any test run. This isn't in a one-time setup fixture method
```
Question, would it make more sense to move out docs tests into separate test class and do this under `[OneTimeSetUp]`?

I'm not gonna be too annoying about this, but we use NUnit for a reason, otherwise every test would have some custom voodoo setup code around it.

PS. Fine merging it any way

andrew-oc replied:
Actually yes it would. Good idea.

andrew-oc committed:
Tests involving docfx moved to their own class

This enabled using OneTimeSetUp for running docfx once, which is much cleaner than the previous way.

andrew-oc committed:
Merge branch 'develop' into improve_ci_performance

andrew-oc committed:
Remove unused variables

andrew-oc committed:
Merge branch 'develop' into improve_ci_performance
